src_sycl/simulator.dp.cpp

```diff
@@ -316,88 +312,51 @@ You may need to rewrite this code.
   /* O(n^2) implementation (no distance threshold), with no shared
      memory etc.
    */
-  void particle_interaction_b(ParticleData_d pPos,
-      ParticleData_d pNextPos,
-      ParticleData_d pVel, SimParam params,
-      const sycl::nd_item<1> &item_ct1) {
-    int id = item_ct1.get_local_id(0) +
-      (item_ct1.get_group(0) * item_ct1.get_local_range(0));
-    if (id >= params.numParticles) return;
-
-    vec3 force(0.0f, 0.0f, 0.0f);
-    vec3 pos(pPos.x[id], pPos.y[id], pPos.z[id]);
+  template <CalculationMethod ct>
+    void particle_interaction(ParticleData_d pPos,
```

Comment:
The formatting looks like it's gone a little funny here - I wouldn't normally indent after the template line.